XYZDriversWebService/src/main/java/com/webservices/xyzdriverswebservice/ClaimEligibility.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.webservices.xyzdriverswebservice;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.jws.WebService;
+import javax.jws.WebMethod;
+import javax.jws.WebParam;
+import javax.ejb.Stateless;
+
+/**
+ *
+ * @author Nathan
+ */
+@WebService(serviceName = "ClaimEligibility")
+@Stateless()
+public class ClaimEligibility {
+
+    /**
+     * Web service operation
+     */
+    @WebMethod(operationName = "Eligibility")
+    public String Eligibility(
+            @WebParam(name = "username") String username,
+            @WebParam(name = "joinedDate") String joinedDate,
+            @WebParam(name = "listOfClaimDates") ArrayList<String> listOfClaimDates,
+            @WebParam(name = "listOfClaimStatuses") ArrayList<String> listOfClaimStatuses) {
+
+        int claimCounter = 0;
+        Calendar calendar = Calendar.getInstance();
+        Calendar calendar2 = Calendar.getInstance();
+        Calendar calendarJoinedDate = Calendar.getInstance();
+        List<Calendar> calendarClaimDates = new ArrayList();
+        
+        try {
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+            Date date = sdf.parse(joinedDate);
+            calendarJoinedDate.setTime(date);
+            
+            for(int i = 0; i < listOfClaimDates.size(); i++)
+            {
+                Calendar tempCalendar = Calendar.getInstance();
+                Date tempDate = sdf.parse(listOfClaimDates.get(i));
+                tempCalendar.setTime(tempDate);
+                calendarClaimDates.add(tempCalendar);
+            }
+            
+            NullCheck(username, calendarJoinedDate, listOfClaimDates, listOfClaimStatuses);
+
+            calendar.add(Calendar.MONTH, -6); //six months ago
+
+            if (calendarJoinedDate.after(calendar)) {
+                return username + " has joined less than six months ago. As such, they are not yet eligible to make a claim.";
+            }
+
+            calendar.set(Calendar.YEAR, 2017); //start of current year

```

Comment:
Can we discover the year dynamically instead of hard-coding it to this year? (I know, harsh!)